src/groups/mwc/mwcio/mwcio_tcpendpoint.cpp

```diff
@@ -90,11 +90,11 @@ bool TCPEndpoint::fromUri(const bsl::string& uri)
     }
 
     // Extract the port part: i.e. after the last ':'
-    d_port = bsl::strtol(uri.c_str() + colon + 1, 0, 10);
-
-    if (d_port == 0) {
+    const long port = bsl::strtol(uri.c_str() + colon + 1, 0, 10);
+    if (port < 0 || port > 65535) {
         return false;  // RETURN
     }
+    d_port = static_cast<int>(port);
```

Comment:
According to `std::strtol` docs ( https://cplusplus.com/reference/cstdlib/strtol/ ) , `strtol` returns 0 either when conversion failed or when conversion succeeded with string `"0"`. These 2 cases are hard to distinguish so I propose to exclude 0 value for simplicity, so my initial comment was not good

```suggestion
    const long port = bsl::strtol(uri.c_str() + colon + 1, 0, 10);
    // For simplicity, do not accept ambiguous `port` value 0
    if (port <= 0 || port > 65535) {
        return false;  // RETURN
    }
    d_port = static_cast<int>(port);
```

`bsl::stoi` implementation uses `strtol`, and you can see the complexity of checks around it:
https://github.com/bloomberg/bde/blob/a30b9d4cb50ade5ea0b2ffded7cff60fa59c0112/groups/bsl/bslstl/bslstl_string.cpp#L193